mindmeld/components/entity_resolver.py

```diff
@@ -562,38 +776,829 @@ def _predict_exact_match(self, entity):
 
         values = []
         for cname in cnames:
-            for item in self._exact_match_mapping["items"][cname]:
+            for item in self._entity_mapping["items"][cname]:
                 item_value = copy.copy(item)
                 item_value.pop("whitelist", None)
                 values.append(item_value)
 
         return values
 
-    def load(self):
-        """Loads the trained entity resolution model from disk."""
-        try:
-            if self._use_text_rel:
-                scoped_index_name = get_scoped_index_name(
-                    self._app_namespace, self._es_index_name
+    def _load(self):
+        self.fit()
+
+
+class SentenceBertCosSimEntityResolver(EntityResolverModelBase):
+    """
+    Resolver class for bert models as described here:
+    https://github.com/UKPLab/sentence-transformers
+    """
+
+    TRANSFORMER_MODEL, POOLING_MODEL, SBERT_MODEL = None, None, None
```

Comment:
Not clear why we have these conditionals

Reply:
Now I see why we use these, why cant these be object attributes? That we, instead of doing this:
```
            setattr(SentenceBertCosSimEntityResolver, "TRANSFORMER_MODEL", transformer_model)
            setattr(SentenceBertCosSimEntityResolver, "POOLING_MODEL", pooling_model)
            setattr(SentenceBertCosSimEntityResolver, "SBERT_MODEL", sbert_model)
```

we do this:
```
self.TRANSFORMER_MODEL = transformer_model
etc
```

Reply:
[redirecting](https://github.com/cisco/mindmeld/pull/278#discussion_r613617506)